### PR TITLE
[smart-switch] Extend config generator for t1-smartswitch topology.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
#### Why I did it
The Smart Switch includes two types of devices (Switch and DPU) that require the sample configuration. Generate Smart Switch sample configurations for Switch or DPU based on the device type.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Extend the sample config generator for t1-smartswitch topology.

#### How to verify it
Run sonic-cfggen UT. 

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-config-engine/config_samples.py [hunk 1]
```diff
@@ -75,25 +77,19 @@ def generate_t1_sample_config(data):
...
-def generate_t1_smartswitch_sample_config(data):
+def generate_t1_smartswitch_switch_sample_config(data, ss_config):
     data = generate_t1_sample_config(data)
     data['DEVICE_METADATA']['localhost']['subtype'] = 'SmartSwitch'
 
     mpbr_prefix = '169.254.200'
     mpbr_address = '{}.254'.format(mpbr_prefix)
 
     bridge_name = 'bridge_midplane'
-    data['MID_PLANE_BRIDGE'] = {
```
Comment (dgsudharsan) [inline src/sonic-config-engine/config_samples.py:86]: Can you please clarify why is this removed?

Comment (oleksandrivantsiv) [reply src/sonic-config-engine/config_samples.py:86]: Based on the changes in IP address assignment HLD, it is no longer needed. This configuration now sits in platform.json and is used only by midplane bridge service

**Review by dgsudharsan — APPROVED** [review]
